cgi-bin/LJ/User/Icons.pm

```diff
@@ -900,4 +900,62 @@ sub _create_mapid {
     return $mapid;
 }
 
+# <LJFUNC>
+# name: LJ::icon_keyword_menu
+# class: web
+# des: Gets all userpics for a given user, separated by keyword. Use this when
+#      building a drop-down icons menu.
+# args: user
+# des-user: a user object.
+# returns: An array of hashrefs like:
+#          {value => ..., text => ..., data => { url => ..., description => ... }}
+#          which can be passed directly to an LJ::html_select(). Includes an
+#          item for each keyword (thus duplicating icons with multiple
+#          keywords), and an item for the default userpic. If no userpics or if
+#          user is undefined, returns an empty array.
+# </LJFUNC>
+sub icon_keyword_menu {
```

Comment:
Why did we move this out of `LJ::Web`? My mental model is that the `LJ::User` classes are the models, and they don't care anything about what the output will be... and separately, we have `LJ::Web` + controllers/TT files which are the views, those actually do transformations.

Maybe it's fine, but I'm curious what the thinking is.

Reply:
Primarily, to simplify the process of using the new icon components. Before, you'd have to either call the function seperately in the controller, or pass in a reference to it as a variable to the template. Now the icon component really just needs the remote, so you don't have to worry about dealing with other dependencies. 

In terms of organization - it generates the icon list for that specific user, and LJ::Web is kind of a chaotic mess with a billion things stuffed in it.